### ISSUE TITLE
Can "result" be list of strings instead of a string?

### ISSUE THREAD

**Issue by peterweb2005** [csdcorp/speech_to_text#23]
thanks

**Comment by sowens-csd** [comment]
What list are you looking for? Each recognition session returns only a single string so I'm not sure what you'd like the contents of the list to be.

**Comment by peterweb2005** [comment]
thanks

previously, i needed the result list of strings and to find which one of the words my program need 

because some words may sound very alike

```
        final ArrayList<String> results = bundle.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
        log.trace("results = {}", results);
```

**Comment by sowens-csd** [comment]
Thanks, I understand now. I'm only returning the highest confidence value (I think) and should be giving back all of the values when available. I'll have a look at adding a different way to access the results in an update to the lib.

**Comment by karincayazilim** [comment]
Hello; I needed a list of results instead of a single result. Google returns the 5 most recent results. is it possible to be this way

**Comment by karincayazilim** [comment]
Is it possible to give in which language he speaks as a parameter? (Locale)

**Comment by sowens-csd** [comment]
Yes, I'm going to be adding a list as an option in the next release of the plugin. 

The locale is an input parameter, I could echo it in the output if that would be helpful? I don't believe that either Android or iOS will detect the language, they have to be told.

**Comment by karincayazilim** [comment]
thanks for your feedback. Although I have set English in some packages, it also perceives me to speak Turkish. I don't want this, but I think it's not possible

**Comment by sowens-csd** [comment]
There's an early version of the ability to retrieve the list of transcriptions on `master` now. If anyone is curious please give it a try. It's not yet used in the example app but it is documented in the change log, and in the API docs. See `SpeechRecognitionResult.alternates` to get started. It should be fully backwards compatible.

This will be released as 0.8.0 as soon as I've done some testing.

**Comment by karincayazilim** [comment]
i waiting  0.8.0

**Comment by Ccastillo06** [comment]
Hey @sowens-csd thanks so much for implementing this feature! This plugin is awesome and pretty easy to implement compared to the ones I tested! 🚀

**Comment by sowens-csd** [comment]
> Hey @sowens-csd thanks so much for implementing this feature! This plugin is awesome and pretty easy to implement compared to the ones I tested! 🚀

Thanks so much for the feedback! Very nice to hear.

**Comment by sowens-csd** [comment]
Released as 0.8.0 now. All feedback welcome.
